app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java

```
@@ -0,0 +1,369 @@
+                    .map(client -> {
+                        client.close();
+                        return true;
+                    })
+                    .defaultIfEmpty(false)
+                    .map(isValid -> BooleanUtils.isTrue(isValid)
+                            ? new DatasourceTestResult()
+                            : new DatasourceTestResult("Unable to create DynamoDB Client.")
+                    );
+        }
+
+    }
+
+    private static String toLowerCamelCase(String action) {
+        return action.substring(0, 1).toLowerCase() + action.substring(1);
+    }
+
+    /**
+     * Given a map that conforms to what a valid DynamoDB request should look like, this function will convert into
+     * a DynamoDBRequest object from AWS SDK. This is done using Java's reflection API.
+     * @param mapping Mapping object representing the request details.
+     * @param type Request type that should be created. Eg., ListTablesRequest.class, PutItemRequest.class etc.
+     * @param <T> Type param of the request class.
+     * @return An object of the request class, containing details of the request from the mapping.
+     * @throws IllegalAccessException Thrown if any of the SDK methods' contracts change.
+     * @throws InvocationTargetException Thrown if any of the SDK methods' contracts change.
+     * @throws NoSuchMethodException Thrown if any of the SDK methods' contracts change.
+     * @throws InstantiationException Thrown if any of the SDK methods' contracts change.
+     */
+    public static <T> T plainToSdk(Map<String, Object> mapping, Class<T> type)
+            throws IllegalAccessException, InvocationTargetException, NoSuchMethodException,
+            AppsmithPluginException {
+        final Class<?> builderType;
+        try {
+            builderType = Class.forName(type.getName() + "$Builder");
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+            return null;
```

Could we either throw the exception or throw a plugin error here itself? We seem to be catching the exception and then returning null.

---

Actually yes, good catch. Done.